Data pipeline

mweiss17 opened the pull request:
# Description

- fixes a bug where pickles for multi-day timeslot updates were not written out
- improves the batch script for running and post-processing experiments 
N.B., built on top of the config_refactor PR



## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

Commit:
merged

Commit:
merge

mweiss17 committed:
moving folders around

mweiss17 committed:
Merge branch 'develop' of github.com:pg2455/covid_p2p_simulation into develop

mweiss17 committed:
exp_config.yml

mweiss17 committed:
moving configs into configs/

mweiss17 committed:
writing a new batch script

mweiss17 committed:
fixing up the config imports

mweiss17 committed:
Merge branch 'config_refactor' into data_pipeline

mweiss17 committed:
added naive_config

mweiss17 committed:
Added the new arg for experimental configs

mweiss17 committed:
Merge branch 'config_refactor' into data_pipeline

mweiss17 committed:
finished batch script and timeslot writing'

mweiss17 committed:
added README to output datasets

Commit:
Merge branch 'data_pipeline' of https://github.com/mweiss17/covid_p2p_simulation into data_pipeline

plstcharles commented:
LGTM; for others: it seems like a lot of changes, but it's because this branch is based off the "config refactor" branch (PR #113 ), and that one should be merged first.

The actual changes for this branch can be seen here:
https://github.com/mweiss17/covid_p2p_simulation/compare/config_refactor...mweiss17:data_pipeline

...this is pretty clean.

Commit:
Merge branch 'config_refactor' of https://github.com/mweiss17/covid_p2p_simulation into mweiss17-config_refactor

Commit:
hhanding off to martin for speed check

mweiss17 committed:
refactoring configs furhter

mweiss17 committed:
timing

mweiss17 committed:
finished refactoring config

mweiss17 committed:
exp_config_path => config

mweiss17 committed:
fixed tests

mweiss17 committed:
cleaning code

mweiss17 committed:
re-adding python workflow

Commit:
merge

Commit:
Merge branch 'config_refactor' of https://github.com/mweiss17/covid_p2p_simulation into data_pipeline

Commit:
new configs from prateek

satyaog committed:
Fix inference not called on some humans

Humans with time slot at hour 0 were sent to the inference server 1 day late

mweiss17 committed:
trying to fix tests

mweiss17 committed:
Merge pull request #18 from satyaog/config_refactor

Fix inference not called on some humans

mweiss17 committed:
Merge branch 'config_refactor' into data_pipeline

mweiss17 committed:
going with tristan's buckets

mweiss17 committed:
remove pdb

satyaog committed:
Fix CI

mweiss17 committed:
Merge pull request #19 from satyaog/data_pipeline

Fix CI

mweiss17 committed:
fixed import

mweiss17 committed:
removing unused try except

satyaog committed:
Fix sending future data to inference server

Humans with time_slot at hour 0 don't have the time to update their data before it gets sent to the inference server. 

Previous fix was wrong by waiting a day to send those data. 

This code is a better fixes it by sending a day old data instead of waiting a day before sending the data

Proper fix would be to update the data at the same time it gets sent to the inference server but that would require a significant refactor of the Human.run() if we want to put it there

satyaog committed:
Fix models_test

Update the test to check the right pkl filename

mweiss17 committed:
Merge pull request #20 from satyaog/data_pipeline

Fix sending future data to inference server

pg2455 commented on src/covid19sim/base.py:
```diff
@@ -240,17 +243,25 @@ def _compute_preferences(self):
 
-            if INTERVENTION_DAY >= 0 and self.current_day == INTERVENTION_DAY:
+            if self.env.timestamp.hour == 0 and self.env.timestamp != self.env.initial_timestamp:
+                self.current_day += 1
```
Why is it required to be here? I think day should increment at the end of the day... more likely after the `yield`

mweiss17 replied:
I think Satya can answer better than I can, but it has to do with the change to multiple updates per day. Basically, people who update in early timeslots are going to have their risk calculated before they've updated their symptoms in the "at" loop.

mweiss17 replied:
Actually, the run loop, here: https://github.com/pg2455/covid_p2p_simulation/pull/116/files#diff-b40966745ead579e546716d7d5c0ee0fR614

satyaog replied:
This is to associate the hour 0 to the current day. The increment of the current day needs to be done before the intervention otherwise hour 0 is always associated to the previous day

pg2455 replied:
But this will not trigger intervention if `INTERVENTION_DAY=0` because `self.current_day`  starts from `1` now.

pg2455 commented on README.md:
```diff
@@ -59,6 +59,7 @@ python run.py test
 @click.option('--seed', help='seed for the process', type=int, default=0)
 @click.option('--n_jobs', help='number of parallel procs to query the risk servers with', type=int, default=1)
 @click.option('--port', help='which port should we look for inference servers on', type=int, default=6688)
+@click.option('--exp_config_path', help='which experiment would we like to run', type=str, default="configs/naive_config.yml")
```
I think it is `config` in the code below

mweiss17 replied:
ah crap, that reverted.

mweiss17 replied:
updated.

pg2455 replied:
💯

mweiss17 committed:
readme update

satyaog committed:
Bring back python 3.8

tensorflow is now an extra in the ctt project allowing to test with python 3.8 in here

mweiss17 committed:
Merge pull request #21 from satyaog/data_pipeline

Bring back python 3.8

satyaog commented:
All tests pass! Should we merge? (btw I've added Olexa trick to ctt to decouple tensorflow from the project so we can again test on python 3.8)

mweiss17 committed:
merged

mweiss17 committed:
Merge branch 'data_pipeline' of github.com:mweiss17/covid_p2p_simulation into data_pipeline

mweiss17 committed:
fixing config import

mweiss17 committed:
tests pass

mweiss17 committed:
hotfix on clustering
